metron-platform/metron-writer/src/main/java/org/apache/metron/writer/hdfs/HdfsWriter.java

```diff
@@ -69,14 +66,19 @@ public void init(Map stormConfig, WriterConfiguration configurations) {
 
 
   @Override
-  public void write(String sourceType
+  public BulkWriterResponse write(String sourceType
                    , WriterConfiguration configurations
                    , Iterable<Tuple> tuples
                    , List<JSONObject> messages
                    ) throws Exception
   {
     SourceHandler handler = getSourceHandler(sourceType);
     handler.handle(messages);
+
+    // HDFS SourceHanlder manages writing and rotating files.  Will throw its own Exceptions.
```

Comment:
typo on `SourceHandler`.

Also, if the `SourceHandler` manages writing and will throw exceptions, do we want to catch the exception, add the tuples as errors to the response and rethrow?